File: frontend/contractor-babu-app/src/types/quotation.ts
export interface QuotationItem {
    id: string;
    quotationId: string;
    description?: string;
    quantity: number;
    area?: number;
    length?: number;
    width?: number;
    height?: number;
    unit: string;
    rate: number;
    amount: number;
    isWithMaterial: boolean;
    sequence: number;
}

export interface Quotation {
    id: string;
    quotationNumber: string;
    projectId: string;
    projectName?: string;
    siteId?: string;
    siteName?: string;
    // Optional client/contact fields
    clientName?: string;
    clientPhone?: string;
    clientEmail?: string;
    clientAddress?: string;
    description?: string;
    remarks?: string;
    quotationDate: string;
    status: string;
    subTotal: number;
    taxPercentage: number;
    taxAmount: number;
    discountPercentage: number;
    discountAmount: number;
    grandTotal: number;
    isActive: boolean;
    items: QuotationItem[];
    createdAt: string;
}

export interface CreateQuotationItemDto {
    id?: string;
    description?: string;
    quantity: number;
    area?: number;
    length?: number;
    width?: number;
    height?: number;
    unit: string;
    rate: number;
    amount: number;
    isWithMaterial: boolean;
    sequence: number;
}

export interface CreateQuotationDto {
    projectId?: string;
    siteId?: string;
    description?: string;
    remarks?: string;
    quotationDate?: string;
    taxPercentage: number;
    discountPercentage: number;
    items: CreateQuotationItemDto[];
}

export interface UpdateQuotationDto {
    description?: string;
    remarks?: string;
    quotationDate?: string;
    taxPercentage?: number;
    discountPercentage?: number;
    status?: string;
    isActive?: boolean;
}

export interface UpdateQuotationItemDto {
    description?: string;
    quantity?: number;
    area?: number;
    length?: number;
    width?: number;
    height?: number;
    unit?: string;
    rate?: number;
    amount?: number;
    isWithMaterial?: boolean;
    sequence?: number;
}

File: backend/CleanArchitectureApp.Domain/Entities/Quotation.cs
namespace CleanArchitectureApp.Domain.Entities;

/// <summary>
/// Represents a BOQ (Bill of Quantities) quotation for a project.
/// Contains all items with quantities and rates.
/// </summary>
public class Quotation : TenantEntity
{
    public string Id { get; set; } = string.Empty;
    public required string QuotationNumber { get; set; }
    public string ProjectId { get; set; } = string.Empty;
    public Project? Project { get; set; }
    
    public string? SiteId { get; set; }
    public Site? Site { get; set; }
    
    public string? Description { get; set; }
    public string? Remarks { get; set; }
    
    public DateTime QuotationDate { get; set; } = DateTime.UtcNow;
    
    public string Status { get; set; } = "Draft"; // Draft, Sent, Approved, Rejected
    
    public decimal SubTotal {  get; set; }
    public decimal TaxPercentage {  get; set; }
    public decimal TaxAmount {  get; set; }
    public decimal DiscountPercentage {  get; set; }
    public decimal DiscountAmount {  get; set; }
    public decimal GrandTotal {  get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Navigation properties
    public ICollection<QuotationItem> Items { get; set; } = new List<QuotationItem>();
}
